support/cas-server-support-pm-ldap/src/main/java/org/apereo/cas/config/LdapPasswordManagementConfiguration.java

```diff
@@ -21,6 +22,7 @@
  * @since 5.2.0
  */
 @Configuration(value = "ldapPasswordManagementConfiguration", proxyBeanMethods = false)
+@ConditionalOnProperty(prefix = "cas.authn.pm.ldap", name = "ldapUrl")
```

Comment:
This will not work. The setting is defined as `cas.authn.pm.ldap[x].ldapUrl` in master.

To make things consistent between master and 6.1, I suggest that if an LDAP url is not defined, a new BeanCreationException should be thrown instead, rather than ignoring the configuration silently.

Reply:
Leaving aside the issue in master, I wanted to remain consistent with what we already have on many classes like:

```java
@ConditionalOnProperty(prefix = "cas.acceptableUsagePolicy", name = "enabled", havingValue = "true", matchIfMissing = true)
public class CasAcceptableUsagePolicyRestConfiguration {

@ConditionalOnProperty(prefix = "cas.audit", name = "enabled", havingValue = "true", matchIfMissing = true)
public class CasCoreAuditConfiguration {

@ConditionalOnProperty(prefix = "cas.events", name = "enabled", havingValue = "true", matchIfMissing = true)
public class CasCoreEventsConfiguration {

...
```

And I think it would be better to have the same behavior everywhere. What do you think?

Reply:
Good point. You're right. To handle this then, it think we need a custom condition with a conditional annotation to loop through the collection of ldaps and skip been creation if none is defined.

Reply:
OK. I'll re-work on that next Tuesday.

Reply:
In fact, I found out a very simple solution to check the whole property name.